gtest/src/manager.rs

```diff
@@ -475,8 +478,62 @@ impl ExtManager {
             .into_parts()
             .0;
 
-        // The `metawasm` macro knows how to decode this as a tuple
-        let mut mapping_code_payload = argument.unwrap_or_default();
+        let metadata =
+            core_processor::informational::execute_for_reply::<WasmiEnvironment<Ext, _>, _>(
+                "metadata".to_string(),
+                mapping_code.clone(),
+                None,
+                None,
+                None,
+                vec![],
+                u64::MAX,
+                self.block_info,
+            )
+            .map_err(TestError::ReadMetadataError)?;
+        let metadata =
+            MetawasmData::decode(&mut &*metadata).expect("fn metadata() returned incorrect data");
```

Comment:
Why so complex? `&*metadata`

Reply:
`decode` takes `&mut Input` as parameter and `Input` is only implemented for `&[u8]` so we need to get `&mut &[u8]` from `Vec<u8>`

Reply:
I'd use [`as_mut_slice`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.as_mut_slice) instead of deref + ref.

Reply:
`as_slice` you mean?